src/firstvoices/backend/models/managers.py

```diff
@@ -0,0 +1,20 @@
+from django.db import models
+
+
+class PermissionsManager(models.Manager):
+    """
+    This custom manager adds additional functionality to filter models based on user permissions. For example the
+    get_viewable_for_user function allows the filtering of model objects by the view permissions a user has.
+    """
+
+    def get_viewable_for_user(self, user):
+        """This function returns a queryset containing model objects that a user has view permissions on."""
+
+        # Get the list of model UUIDs that the user has view permissions for
+        models_uuid_list = [
+            model.id
+            for model in self.get_queryset()
+            if user.has_perm(f"backend.view_{model.__class__.__name__.lower()}", model)
```

Comment:
I just realized that rules has a convenient function for getting the permission name-- this could be `user.has_perm(model.get_perm('view'))`. See: https://github.com/dfunckt/django-rules#permissions-in-models

Reply:
Good catch, thanks. I have updated the check to use the function. Note that I still had to pass the model object so that the predicates have access to it: `user.has_perm(model.get_perm("view"), model)`